cassandra/connection.py

```diff
@@ -360,6 +368,28 @@ def error_all_requests(self, exc):
                             "failed connection (%s) to host %s:",
                             id(self), self.host, exc_info=True)
 
+        # run first callback from this thread to ensure pool state before leaving
+        cb, _ = requests.popitem()[1]
+        try_callback(cb)
+
+        if not requests:
+            return
+
+        # additional requests are optionally errored from a separate thread
+        # The default callback and retry logic is fairly expensive -- we don't
+        # want to tie up the event thread when there are many requests
+        def err_all_callbacks():
+            for cb, _ in requests.values():
+                try_callback(cb)
+        if len(requests) < Connection.CALLBACK_ERR_THREAD_THRESHOLD:
+            err_all_callbacks()
+        else:
+            # daemon thread here because we want to stay decoupled from the cluster TPE
+            # TODO: would it make sense to just have a driver-global TPE?
+            t = Thread(target=err_all_callbacks)
```

Comment:
It's a good thing to consider. I thought about it and here is my reasoning:
- nodes down should be relatively infrequent, and thread overhead is not a concern
- if there is a sudden loss of connection, the connection errors will be processed serially by the event thread
- only <node count> threads *may* be created
- clusters with higher node count are very unlikely to have high concurrent requests per connection (thus the threshold for starting a thread will not be breached)